modules/imgproc/src/peronamalik.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "precomp.hpp"
+#include "opencl_kernels_imgproc.hpp"
+#include "cvconfig.h"
+
+#include <iostream>
+using namespace std;
+
+namespace cv
+{
+    void PeronaMalik(InputArray src, OutputArray dst, double timeStepSize, double k, int noOfTimeSteps, int fluxFunc )
+    {
+        CV_Assert(k != 0);
+        CV_Assert(timeStepSize > 0);
+        CV_Assert(noOfTimeSteps >= 0);
+        CV_Assert((fluxFunc == CV_PERONA_MALIK_EXPONENTIAL) || (fluxFunc == CV_PERONA_MALIK_INVERSE_QUADRATIC));
+        CV_Assert(src.size() == dst.size());
+        CV_Assert(src.type() == dst.type());
+        
+        Mat _dst = dst.getMat();
+        Mat _src = src.getMat().clone();
+        
+        int channels = _src.channels();
+        int nRows = _src.rows;
+        int nCols = _src.cols * channels;
+        
+        double oneOnkSquared = 1 / (k * k);
+        
+        int i,j;
+        for (int t = 0; t < noOfTimeSteps; t++)
+        {
+            for( i = 0; i < nRows; ++i)
```

Comment:
the loop should be very easy to parallelize; you are welcome to do that, see for example, opencv/modules/imgproc/src/smooth.cpp, bilateral filter implementation for details